app/services/resource_service.rb

```diff
@@ -80,4 +82,18 @@ def resource_or_error_message(record)
       GraphQL::ExecutionError.new("Invalid input: #{record.errors.full_messages.join(", ")}")
     end
   end
+
+  def set_default_categories
+    setting = data_provider.data_resource_settings.where(data_resource_type: @resource_class.name).first
+    return if setting.blank?
+    return if setting.default_category_ids.blank?
+
+    categories_to_add = setting.default_category_ids.compact.uniq.delete_if(&:blank?)
+    return if categories_to_add.blank?
+
+    categories_to_add.each do |cat_id|
+      category = Category.find_by(id: cat_id)
```

Comment:
in den anderen Dateien schreibst du `category_to_add = Category.where(name: category_name).first_or_create` und hier `category = Category.find_by(id: cat_id)`.
könnte hier auch `category_to_add = Category.where(id: cat_id).first` sein dann, oder wieso ist an der stelle `find_by` besser? zumindest einheitliches `category` oder `category_to_add` verwenden?

Reply:
first_or_create ist der Unterschied, beim einen will ich es nur raussuchen und an der anderen Stelle wird es auch angelegt, wenn es vorher nicht existiert hatte.

`Category.where(id: cat_id).first` wäre identisch zu `Category.find_by(id: cat_id)`

`.where.first` suggeriert irgendwie, das es mehrere geben könnte und man in dem Fall den erste nimmt, falls es mehrere gibt. Bei `find_by()` ist irgendwie klarer das es ein konkreter Datensatz ist den man wählt. `find:by` kann aber nicht mit `first_or_create` kombiniert werden